### PR TITLE
Snapshot related backports

### PR DESCRIPTION
**Pull request by mkleen**
## Summary of the changes / Why this improves CrateDB

See individual commits.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/io/crate/replication/logical/repository/LogicalReplicationRepository.java [hunk 1]
```diff
@@ -255,8 +261,12 @@ public void getRepositoryData(ActionListener<RepositoryData> listener) {
...
                     }
                 }
-                var repositoryData = RepositoryData.EMPTY
-                    .addSnapshot(SNAPSHOT_ID, SnapshotState.SUCCESS, Version.CURRENT, shardGenerations.build());
+                var repositoryData = RepositoryData.EMPTY.addSnapshot(SNAPSHOT_ID,
+                                                                      SnapshotState.SUCCESS,
+                                                                      Version.CURRENT,
+                                                                      shardGenerations.build(),
+                                                                      null,
+                                                                      null);
```
Comment (mkleen) [inline server/src/main/java/io/crate/replication/logical/repository/LogicalReplicationRepository.java:269]: See https://github.com/opensearch-project/cross-cluster-replication/blob/main/src/main/kotlin/org/opensearch/replication/repository/RemoteClusterRepository.kt#L201-L202 as reference.

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 2]
```diff
@@ -1310,7 +1342,7 @@ private RepositoryData getRepositoryData(long indexGen) {
             try (InputStream blob = blobContainer().readBlob(snapshotsIndexBlobName);
                  XContentParser parser = XContentType.JSON.xContent().createParser(NamedXContentRegistry.EMPTY,
                      LoggingDeprecationHandler.INSTANCE, blob)) {
-                return RepositoryData.snapshotsFromXContent(parser, indexGen, true);
+                return RepositoryData.snapshotsFromXContent(parser, indexGen, false);
```
Comment (mkleen) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:1345]: The cached repository data from the original pr uses also false.

File: server/src/testFixtures/java/org/elasticsearch/repositories/blobstore/BlobStoreTestUtil.java [hunk 3]
```diff
@@ -200,48 +218,42 @@ private static void assertSnapshotUUIDs(BlobStoreRepository repository, Reposito
...
-                                    assertThat(shardPathContents.keySet().stream()
-                                                   .filter(name -> name.startsWith(BlobStoreRepository.INDEX_FILE_PREFIX)).count(), lessThanOrEqualTo(2L));
-                                }
-                            }
-                        }
-                    },
-                    (e) -> {}
-                )
-            );
+            SnapshotInfo snapshotInfo = repository.getSnapshotInfo(snapshotId).get();
```
Comment (mkleen) [inline server/src/testFixtures/java/org/elasticsearch/repositories/blobstore/BlobStoreTestUtil.java:221]: Blocking call `.get` on a Future. This is only for testing therefore i think it's acceptable.

**Review by matriv — APPROVED** [review]
LGTM, Left a comment regarding changes in indentation on the 3rd commit (the large one)

File: server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java [hunk 4]
```diff
@@ -536,73 +538,73 @@ private void processExternalChanges(boolean changedNodes, boolean startShards) {
         clusterService.submitStateUpdateTask("update snapshot after shards started [" + startShards +
                 "] or node configuration changed [" + changedNodes + "]", new ClusterStateUpdateTask() {
 
-            private final Collection<SnapshotsInProgress.Entry> finishedSnapshots = new ArrayList<>();
+                    private final Collection<SnapshotsInProgress.Entry> finishedSnapshots = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (mkleen) [reply server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java:540]: checkstyle passed, but i will double check. thanks!

[the remark this answers is not in the document]
Comment (matriv) [reply server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java:540]: Checkstyle doesn't check lucene and elasticsearch packages.

[the remark this answers is not in the document]
Comment (mkleen) [reply server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java:540]: Hm, checkstyle insists on the additional indention here.

[the remark this answers is not in the document]
Comment (matriv) [reply server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java:540]: It seems that the indentation was changed here but not in the original es commit.

[the remark this answers is not in the document]
Comment (matriv) [reply server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java:540]: Apologies, I compared the files (not in github diff view) andyou're right, it's correct as is.

[the remark this answers is not in the document]
Comment (mkleen) [reply server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java:540]: 👍

[the remark this answers is not in the document]
Comment (mkleen) [reply server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java:540]: ok, i found the problem. the indention should have been adopted already on https://github.com/crate/crate/pull/13250/commits/3c3233e8528bbb9838327808301e2537571e4d90

[the remark this answers is not in the document]
Comment (mkleen) [reply server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java:540]: i will fix this.

**Comment by mkleen** [comment]
@Mergifyio requeue

**Pull request merged**